# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [4]:
data.info()
data.describe()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

education       ofjobs     yearsexp       honors    volunteer  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      3.618480     3.661396     7.842916     0.052772     0.411499   
std       0.714997     1.219126     5.044612     0.223601     0.492156   
min       0.000000     1.000000     1.000000     0.000000     0.000000   
25%       3.000000     3.000000     5.000000     0.000000     0.000000   
50%       4.000000     4.000000     6.000000     0.000000     0.000000   
75%       4.000000     4.000000     9.000000     0.000000     1.000000   
max       4.000000     7.000000    44.000000     1.000000     1.000000   

          military     empholes  occupspecific   occupbroad  workinschool  \
count  4870.000000  4870.000000    4870.000000  4870.000000   4870.000000   
mean      0.097125     0.448049     215.637782     3.481520      0.559548   
std       0.296159     0.497345     148.127551     2.038036      0.496492   
min       0.000000     0.000000       7.000000     1.000000      0.000000   
25%       0.000000     0.000000      27.000000     1.000000      0.000000   
50%       0.000000     0.000000     267.000000     4.000000      1.000000   
75%       0.000000     1.000000     313.000000     6.000000      1.000000   
max       1.000000     1.000000     903.000000     6.000000      1.000000   

          ...           educreq      compreq       orgreq        manuf  \
count     ...       4870.000000  4870.000000  4870.000000  4870.000000   
mean      ...          0.106776     0.437166     0.072690     0.082957   
std       ...          0.308866     0.496083     0.259649     0.275854   
min       ...          0.000000     0.000000     0.000000     0.000000   
25%       ...          0.000000     0.000000     0.000000     0.000000   
50%       ...          0.000000     0.000000     0.000000     0.000000   
75%       ...          0.000000     1.000000     0.000000     0.000000   
max       ...          1.000000     1.000000     1.000000     1.000000   

          transcom     bankreal        trade   busservice   othservice  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      0.030390     0.085010     0.213963     0.267762     0.154825   
std       0.171677     0.278932     0.410141     0.442847     0.361773   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     1.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

           missind  
count  4870.000000  
mean      0.165092  
std       0.371308  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000  

[8 rows x 55 columns]

In [5]:
data.groupby('race').count()
# Same number of resumes submitted for black and white sounding names

id    ad  education  ofjobs  yearsexp  honors  volunteer  military  \
race                                                                         
b     2435  2435       2435    2435      2435    2435       2435      2435   
w     2435  2435       2435    2435      2435    2435       2435      2435   

      empholes  occupspecific    ...      compreq  orgreq  manuf  transcom  \
race                             ...                                         
b         2435           2435    ...         2435    2435   2435      2435   
w         2435           2435    ...         2435    2435   2435      2435   

      bankreal  trade  busservice  othservice  missind  ownership  
race                                                               
b         2435   2435        2435        2435     2435       2435  
w         2435   2435        2435        2435     2435       2435  

[2 rows x 64 columns]

In [6]:
# number of callbacks for black-sounding names
print(sum(data[data.race=='b'].call))
# number of callbacks for white-sounding names
print(sum(data[data.race=='w'].call))

157.0
235.0


1. What test is appropriate for this problem? Does CLT apply?
Binomial distribution appears to be best as we have number of call backs which relates to trials with a probability of success given by call backs due to black or white sounding name. The call backs are independent of one another as the resumes are not dependent on one another and  the same number of black sounding and white sounding named resumes were sent. CLT therefore applies.

2. What are the null and alternate hypotheses?
The null hypothesis is that having a black or white sounding name does not impact call back rates from employers. The alternate hypothesis is that having a black sounding name does impact call back rates from emplyoers compared to having a white sounding name

In [13]:
# Under null hypothesis with black and white sounding names being the same then
# the chance is 50/50 for both groupings with 392 trials. The number of resumes
# submitted were the same for both groupings so that is eliminated as a variable
samples = np.random.binomial(392, 0.5, size=10000)

conf_int = np.percentile(samples, [2.5, 97.5])
print('95% confidence interval is: ',conf_int)

95% confidence interval is:  [ 177.  216.]


In [14]:
margin_of_error = 215 - 398/2
print('Margin of Error at 95% confidence inverval is: ', margin_of_error)

Margin of Error at 95% confidence inverval is:  16.0


In [19]:
p = np.sum(samples >= 235) / len(samples)
print('P-value is ', p)

P-value is  0.0


3. Compute margin of error, confidence interval, and p-value.
4. Write a story describing the statistical significance in the context of the original problem.

4870 resumes were submitted in response to over 1300 newspaper ads for sales, administrative and clerical jobs in Boston and Chicago. 2435 Resumes were randomly assigned a black sounding name (e.g. Lakisha Washington, Jamal Jones). 2435 resumes were randomly assgined a white sounding name (e.g. Emily Walsh or Brendan Baker). Names were chosen according to frequency data obtained from brith certificates of Massachusetts births from 1974 and 1979 and the blackness or whiteness was confirmed by surveys in public areas in Chicago. Resumes variety by quality. 2 high and low quality resumes were sent to each job opening. 

235 with with white sounding names were called back for an interview while 157 with black sounding names were called back from an interview. Our null hypothesis is that having a black or white sounding name should not impact call back rates or assumes a 50/50 chance of call back. Our P-value of 0 shows that it is highly unlikely we would observe our sample result if the null hypothesis is true. 

5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

No the analysis does not mean race/name is the most important factor in callback success. The analysis shows that how the names were classified as black and white sounding is statistically significant on callback success. It does not necessitate that it is the only factor let alone the most important factor. 

This being said it does show that having a black sounding name results in a lower likelihood of being called back for a position compared to having a white sounding name. 